File: src/main/java/me/apqx/algorithm/sort/BubbleSort.java
package me.apqx.algorithm.sort;

import java.util.Arrays;

public class BubbleSort {

    /**
     * 冒泡排序，升序
     */
    public static void bubbleSort(int[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            boolean flag = false;
            for (int j = 0; j < array.length - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                    flag = true;
                }
            }
            if (!flag) break;
        }
        System.out.println(Arrays.toString(array));
    }

    /**
     * 冒泡排序
     *
     * @param up 是否升序
     */
    static void bubbleSort(int[] array, boolean up) {
        if (up) {
            bubbleSort(array);
        } else {
            for (int i = 0; i < array.length - 1; i++) {
                boolean flag = false;
                for (int j = 0; j < array.length - i - 1; j++) {
                    if (array[j] < array[j + 1]) {
                        int temp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = temp;
                        flag = true;
                    }
                }
                if (!flag) break;
            }
            System.out.println(Arrays.toString(array));
        }
    }
}

File: src/main/kotlin/me/apqx/algorithm/sort/BubbleSort.kt
package me.apqx.algorithm.sort

/**
 * 冒泡排序，升序
 */
fun bubbleSort(array: Array<Int>) {
    for (i in 0 until array.size - 1) {
        var flag = false
        for (j in 0 until array.size - i - 1) {
            if (array[j] > array[j + 1]) {
                var temp = array[j]
                array[j] = array[j + 1]
                array[j + 1] = temp
                flag = true
            }
        }
        if (!flag) break
    }
    println(array.toList())
}

/**
 * 冒泡排序
 *
 * @param up 是否升序
 */
fun bubbleSort(array: Array<Int>, up: Boolean) {
    if (up) {
        bubbleSort(array)
    } else {
        for (i in 0 until array.size - 1) {
            var flag = false
            for (j in 0 until array.size - i - 1) {
                if (array[j] < array[j + 1]) {
                    var temp = array[j]
                    array[j] = array[j + 1]
                    array[j + 1] = temp
                    flag = true
                }
            }
            if (!flag) break
        }
    }
    println(array.toList())
}